src/tail/session.rs

```diff
@@ -28,18 +28,25 @@ impl Session {
         user: GlobalUser,
         shutdown_rx: Receiver<()>,
         tx: Sender<()>,
+        metrics_port: u16,
     ) -> Result<(), failure::Error> {
         // We need to exit on a shutdown command without waiting for API calls to complete.
         tokio::select! {
             _ = shutdown_rx => { Ok(()) }
-            result = Session::start(target, user, tx) => { result }
+            result = Session::start(target, user, tx, metrics_port) => { result }
         }
     }
 
-    async fn start(target: Target, user: GlobalUser, tx: Sender<()>) -> Result<(), failure::Error> {
+    async fn start(
+        target: Target,
+        user: GlobalUser,
+        tx: Sender<()>,
+        metrics_port: u16,
+    ) -> Result<(), failure::Error> {
         let client = http::cf_v4_api_client_async(&user, HttpApiClientConfig::default())?;
 
-        let url = get_tunnel_url().await?;
+        // TODO: make Tunnel struct responsible for getting its own port!
```

Comment:
want to make an issue or are you getting in this pr?

Reply:
not happening in this PR, i'm going to do it as part of #1200